test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,55 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyone()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        AsyncManualResetEvent mainThreadBlockerEvent = new AsyncManualResetEvent(false);
+        AsyncManualResetEvent backgroundThreadMonitorEvent = new AsyncManualResetEvent(false);
+
+        // Start task to monitor IsMainThreadBlockedByAnyone
+        Task.Run(async () =>
+        {
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+            await mainThreadBlockerEvent.WaitAsync();
+
+            bool shouldWaitForMainThreadBlock = true;
+            while (shouldWaitForMainThreadBlock)
+            {
+                try
+                {
+                    Assert.True(this.Context.IsMainThreadBlockedByAnyone);
+                    shouldWaitForMainThreadBlock = false;
+                }
+                catch
+                {
+                    // Give the main thread time to enter a blocking state
+                    Thread.Sleep(50);
+                }
+            }
+
+            backgroundThreadMonitorEvent.Set();
+        });
+
+        JoinableTask? joinable = this.Factory.RunAsync(async delegate
+        {
+            Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+            await Task.Yield();
```

Comment:
I guess this test assumes the test is started on the main thread. I don't remember if this is true, but it would be good to document that in a code comment.
Also, we should have a negative test, where the thread being blocked is _not_ the main thread, and verify that the new property returns false.

Reply:
Added a negative test.